node/pkg/watchers/evm/watcher.go

```diff
@@ -616,6 +622,7 @@ func (w *Watcher) Run(parentCtx context.Context) error {
 					currentEthHeight.WithLabelValues(w.networkName).Set(float64(blockNumberU))
 					stats.Height = int64(blockNumberU)
 					w.updateNetworkStats(&stats)
+					w.ccqAddLatestBlock(logger, ev)
```

Comment:
After much offline discussion, the decision was made to switch the batch poller to subscribe for `newHead` for latest so that rollbacks will be published as new blocks. Also `AddLatest` was reworked to truncate the cache and apply the new block rather than throwing an error on rollback.